snuba/migrations/groups.py

```diff
@@ -257,25 +255,68 @@ def get_migrations(self) -> Sequence[str]:
         return ["0001_functions"]
 
 
-class GenericMetricsLoader(DirectoryLoader):
-    def __init__(self) -> None:
-        super().__init__("snuba.snuba_migrations.generic_metrics")
+HARDCODED_MIGRATIONS = {
+    "SYSTEM": "system",
+    "EVENTS": "events",
+    "TRANSACTIONS": "transactions",
+    "DISCOVER": "discover",
+    "OUTCOMES": "outcomes",
+    "METRICS": "metrics",
+    "SESSIONS": "sessions",
+    "QUERYLOG": "querylog",
+    "PROFILES": "profiles",
+    "FUNCTIONS": "functions",
+    "REPLAYS": "replays",
+}
 
-    def get_migrations(self) -> Sequence[str]:
-        return [
-            "0001_sets_aggregate_table",
-            "0002_sets_raw_table",
-            "0003_sets_mv",
-            "0004_sets_raw_add_granularities",
-            "0005_sets_replace_mv",
-            "0006_sets_raw_add_granularities_dist_table",
-            "0007_distributions_aggregate_table",
-            "0008_distributions_raw_table",
-            "0009_distributions_mv",
-        ]
+CONFIG_BUILT_MIGRATIONS = {
+    loader.migration_group_name: loader
+    for loader in [
+        ConfigurationLoader(config_file)
+        for config_file in glob(settings.MIGRATION_CONFIG_FILES_GLOB, recursive=True)
+    ]
+}
 
 
-_REGISTERED_GROUPS = {
+class _MigrationGroup(type):
+    """
+    This class allows fetching arbitrary static attributes on whichever class extends it. This mimics some of the behaviour of the Enum class.
+    """
+
+    def __getattr__(cls, attr: str) -> "MigrationGroup":
+        if (
+            attr not in HARDCODED_MIGRATIONS
+            and attr.lower() not in CONFIG_BUILT_MIGRATIONS
+        ):
+            raise AttributeError(
+                f"type object 'MigrationGroup' has no attribute '{attr}'"
+            )
+
+        return MigrationGroup(attr.lower())
+
+    def __iter__(cls) -> Iterator[MigrationGroup]:
+        return iter(REGISTERED_GROUPS)
+
+
+class MigrationGroup(metaclass=_MigrationGroup):
+    """
+    This class is a replacement for the Enum class that used to define the different migration groups. In order to avoid having to go through the entire codebase to find instances of MigrationGroup and replace them, this will mimic the functionality of an Enum class while allowing arbitrary values to be populated.
```

Comment:
nit: can you break this docstring up over multiple lines?

Reply:
I don't think this enum like interface is the right one anymore since Migration groups can be defined on the fly. It made sense previously because all of the members were defined in code. The rest of the codebase never references MigrationGroup (only a handful of places within the migrations module) so there are not that many places to change.